### PR TITLE
Rename physical GPU ID to platform GPU ID

### PR DESCRIPTION
**Pull request by whchung**
Fix the 3-tier notion of "GPU ID" for ROCm

- TF GPU ID
- CUDA GPU ID -> incorrectly set as "physical GPU ID" -> platform GPU ID
- physical GPU ID

### CODE REVIEW

**Comment by whchung** [comment]
@sunway513 this fix deserves to be cherry-picked into `r1.8-rocm` once validated with CI

**Comment by whchung** [comment]
unit tests fail with "failed to create session". looking into it.

File: tensorflow/core/common_runtime/gpu/gpu_id_manager_test.cc [hunk 1]
```diff
@@ -30,30 +30,30 @@ PhysicalGpuId TfToPhysicalGpuId(TfGpuId tf) {
```
Comment (deven-amd) [inline tensorflow/core/common_runtime/gpu/gpu_id_manager_test.cc]: Do we need to delete (or atleast update) the routine at lines 25-28? 
It does not seem like it is used at all, but has references `PhysicalGpuId` (guessing this test is not a part of CI, else would have expected this to trigger a compile failure)

Comment (whchung) [reply tensorflow/core/common_runtime/gpu/gpu_id_manager_test.cc]: thanks. let me revise it.

**Pull request merged**